make-util.js

```diff
@@ -1895,65 +1895,49 @@ exports.generateTasks = generateTasks;
 
     if (!originalFunction || originalFunction instanceof Function === false) throw Error('originalFunction is not defined');
     // If the task is building on the ci, we don't want to sync files
     if (callGenTaskDuringBuild === false) return originalFunction;
 
-    console.log(
-        "Syncing generated files with source task...\n" +
-        "----------------------------------------------\n" +
-        "Getting list of uncommitted changes");
-    const initialDiffOutput = run(`git -C "${repoPath}" diff --name-only`)
-    console.log(
-        "uncommitted changes:\n" + 
-        `${initialDiffOutput}\n` +
-        "----------------------------------------------");
-
     return function(taskName, ...args) {
-        // git diff --name-only return "/"" as separator
         originalFunction.apply(this, [taskName, ...args]);
 
-        const genTaskPath = `${genTaskBasePath}/${taskName}/`;
+        const genTaskPath = path.join(basicGenTaskPath, taskName);
+
         // if it's not a generated task, we don't need to sync files
         if (!fs.existsSync(genTaskPath)) return;
+        const [baseTaskName, config] = taskName.split("_");
 
-        const changedPaths = [];
-        const afterTaskBuildDiff = run(`git diff --name-only`);
-
-        afterTaskBuildDiff.split("\n").forEach((line) => {
-            // We skip files that are not in the generated task folder or are already in the initial diff   
-            if (!line.includes(genTaskPath) || initialDiffOutput.includes(line)) return;
-            changedPaths.push(line);
-        });
-
-        if (changedPaths.length === 0) return;
-
-        console.log(
-            "The following generated files where updated in source task after build and added to staged, please commit them to the repo:\n" +
-            `${changedPaths}\n` +
-            "----------------------------------------------\n" +
-            "The following files were copied to source task:");
-
-        changedPaths.forEach((filePath) => {
-            const fileName = path.basename(filePath);
-            if (allowedFilesToCopy.indexOf(fileName) === -1) return;
-            const [baseTaskName, config] = taskName.split("_");
+        runtimeChangedFiles.forEach((fileName) => {
+            const filePath = path.join(genTaskPath, fileName);
+            if (!fs.existsSync(filePath)) return;
+            
             let dest = path.join(__dirname, 'Tasks', baseTaskName, fileName);
             if (config) {
                 dest = path.join(__dirname, 'Tasks', baseTaskName, '_buildConfigs', config, fileName);
             }
 
-            fs.copyFileSync(filePath, dest);
-            console.log(`from ${filePath} to ${dest}`);
+            // Prefer execSync instead of run, because gir diff throw error if there is a diff between files
+            try {
+                ncp.execSync(`git diff --no-index ${filePath} ${dest} --numstat`);
```

Comment:
>t diff [](http://example.com/codeflow?start=32&length=6)

minor: I'm thinking the diff isn't necessary and we can just copy; if the files are identical, git will ignore the change